src/views/ViewAllExplorer.tsx

```diff
@@ -27,11 +25,21 @@ function ViewAllExplorer({
   const { isDataLoaded } = useSnapshot(OptionsCtrl.state);
   const { pairingUri } = useSnapshot(WcConnectionCtrl.state);
   const { themeMode } = useSnapshot(ThemeCtrl.state);
-  const { wallets, recommendedWallets } = useSnapshot(ExplorerCtrl.state);
+  const { wallets } = useSnapshot(ExplorerCtrl.state);
   const shouldLoadWallets = wallets.listings.length === 0;
   const [walletsLoading, setWalletsLoading] = useState(false);
   const loading = !isDataLoaded || !pairingUri || walletsLoading;
-  const _wallets = [...recommendedWallets, ...wallets.listings];
+  const [search, setSearch] = useState('');
+  const timer = useRef<NodeJS.Timeout | null>(null);
+
+  const onChangeText = (value: string) => {
```

Comment:
I believe you are "debouncing" here? Perhaps good candidate to make into util?

Reply:
done! I moved the logic to a new hook